microsoft-azure-storage/src/com/microsoft/azure/storage/core/Canonicalizer.java

```diff
@@ -251,6 +256,11 @@ protected static String getCanonicalizedResource(final java.net.URL address, fin
         final StringBuilder canonicalizedResource = new StringBuilder(resourcepath.toString());
 
         // query parameters
+        if (address.getQuery()!= null  && !address.getQuery().contains("=")) {
+            //no query params. Safe to return empty
+            return canonicalizedResource.toString();
+        }
```

Comment:
Thanks for mentioning this. This should be ideally be (address.getQuery == null || !address.getQuery().contains("=")) to avoid 3 hashmap construction & an arraylist. I will fix this in the next PR to dev branch.